tinker/static/assets/js/tinker.js

```diff
@@ -14,4 +14,348 @@ $(document).ready(function () {
+                if (limitList[i] == numberOfItems) {
+                     $("#selected-option").append("<option value=\" " + limitList[i] + " \" selected='selected'>All</option>");
+                } else {
+                    $("#selected-option").append("<option value=\" " + limitList[i] + " \" selected='selected'>" + limitList[i] + "</option>");
+                }
+            } else {
+                if (limitList[i] == numberOfItems) {
+                    $("#selected-option").append("<option value=\" " + limitList[i] + " \">All</option>");
+                } else {
+                    $("#selected-option").append("<option value=\" " + limitList[i] + " \">" + limitList[i] + "</option>");
+                }
+            }
+        }
+
+        $("#selected-option").change(function() {
+            limitPerPage = $("#selected-option").children("option:selected").attr("value");
+            if (type == 'e-announcement') {
+                Cookies.set('e-announcement-cookie', limitPerPage, { expires: 30, path: '/' });
+            } else if (type == 'event') {
+                Cookies.set('event-cookie', limitPerPage, { expires: 30, path: '/' });
+            }
+
+            $(".pagination").children("li.temp-button").remove();
+
+            let totalPages = Math.ceil(numberOfItems / limitPerPage);
+            maxPages = totalPages;
+
+            createButtons(totalPages, limitPerPage);
+
+            // When a new limit is selected, we reset the current page to one, since if we have 10 pages, and then
+            // we changed to only having one page, we can't keep whatever page we were previously on (if it was greater
+            // than 1) so we instead just set it to 1.
+            let currentPage = 1;
+
+            let grandTotal = limitPerPage * currentPage;
+
+            for (let i = grandTotal - limitPerPage; i < grandTotal ; i ++) {
+                $("#loop .items-to-paginate:eq(" + i + ")").show();
+            }
+
+            switchPageClick(limitPerPage);
+
+            nextOrPreviousPage(limitPerPage, "next-page");
+
+            goToPage(limitPerPage);
+        });
+
+        let totalPages = Math.ceil(numberOfItems / limitPerPage);
+        maxPages = totalPages;
+
+        createButtons(limitPerPage);
+
+        switchPageClick(limitPerPage);
+
+        nextOrPreviousPage(limitPerPage, "next-page");
+
+        nextOrPreviousPage(limitPerPage, "previous-page");
+
+        goToPage(limitPerPage);
+    }
+}
+// Sets the next or previous page depending on what button you pressed
+function nextOrPreviousPage(limitPerPage, type) {
+    $("#" + type + " ").on("click", function () {
+        let currentPage = $(".pagination li.active").index();
+        let totalPages = maxPages;
+
+        if (totalPages > paginationRange) {
+            // Increments or decrements the currentPage depending on which button was pressed
+            if (type === "next-page") {
+                if (currentPage === totalPages + 3) {
```

Comment:
This if is checking to make sure that if someone hits the next page button, we don't go beyond the range of pages and go off into wonderland. It is similar to why I have a check in previous page that says ```if (currentPage === 1 )``` since if current page is equal to 1 and we hit previous, we don't want to go to page 0 (since it doesn't exist) similarly we don't want to go over the max number of pages.

Reply:
Also added a comment about it in the code